src/app/Plans/components/helpers.ts

```diff
@@ -1,122 +1,175 @@
+  return condition ? condition.type : '';
 };
 
-// TODO maybe generalize this for cold migrations too
-type WarmPlanState =
-  | 'NotStarted'
-  | 'Starting'
-  | 'Copying'
-  | 'AbortedCopying'
-  | 'StartingCutover'
-  | 'Cutover'
-  | 'Finished';
-
-export const getWarmPlanState = (
+export const getMigStatusState = (state: PlanState | null, isWarmPlan: boolean) => {
+  let title = '';
+  let variant: ProgressVariant | undefined;
+
+  switch (true) {
+    case state === 'Starting': {
+      title = 'Running';
+      break;
+    }
+    case state === 'NotStarted': {
+      title = '';
+      break;
+    }
+    case state === 'Finished-Failed':
+    case state === 'AbortedCopying': {
+      title = 'Failed';
+      variant = ProgressVariant.danger;
+      break;
+    }
+    case state === 'Finished-Succeeded': {
+      title = 'Succeeded';
+      variant = ProgressVariant.success;
+      break;
+    }
+    case state === 'Copying':
+    case state === 'PipelineRunning': {
+      title = isWarmPlan ? 'Running cutover' : 'Running';
+      break;
+    }
+    case state === 'Finished-Incomplete': {
+      title = 'Finished - Incomplete';
+      variant = ProgressVariant.warning;
+      break;
+    }
+    default: {
+      title = '';
+    }
+  }
+
+  return {
+    title,
+    variant,
+  };
+};
+
+export const getButtonState = (state: PlanState | null): PlanActionButtonType | null => {
+  let type: PlanActionButtonType | null = null;
+
+  switch (true) {
+    case state === 'NotStarted': {
+      type = 'Start';
+      break;
+    }
+    case state === 'Starting': {
+      type = null;
+      break;
+    }
+    case state === 'Copying': {
+      type = 'Cutover';
+      break;
+    }
+    case state === 'Finished-Incomplete':
+    case state === 'Finished-Failed':
+    case state === 'Canceled':
+    case state === 'AbortedCopying': {
+      type = 'Restart';
+      break;
+    }
+    case state === 'StartingCutover': {
+      type = null;
+      break;
+    }
+    case state === 'PipelineRunning': {
+      type = null;
+      break;
+    }
+    case state === 'Finished-Succeeded': {
+      type = null;
+      break;
+    }
+    default: {
+      type = null;
+    }
+  }
+
+  return type;
+};
+
+export const getPlanState = (
   plan: IPlan | null,
-  migration: IMigration | null,
-  migrationsQuery: UseQueryResult<IKubeList<IMigration>>
-): WarmPlanState | null => {
-  if (!plan || !plan.spec.warm) return null;
+  migration: IMigration | null
+): PlanState | null => {
+  if (!plan) return null;
+
   if (!migration) return 'NotStarted';
-  if (isPlanBeingStarted(plan, migration, migrationsQuery)) {
-    return 'Starting';
-  }
+
   const conditions = plan.status?.conditions || [];
+
   if (
-    (hasCondition(conditions, PlanStatusType.Canceled) ||
-      hasCondition(conditions, PlanStatusType.Failed)) &&
+    (hasCondition(conditions, 'Canceled') || hasCondition(conditions, 'Failed')) &&
```

Comment:
AbortedCopying is only relevant for warm plans. This will incorrectly put cold canceled/failed plans in the AbortedCopying state because only warm plans can ever have `migration.spec.cutover`.